engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereAllFilter.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.rowset.WritableRowSet;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.TableDefinition;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A Select filter that always returns the selection.
+ */
+public final class WhereAllFilter extends WhereFilterImpl {
```

Comment:
We already have WhereNoneFilter, but I've gone ahead and deleted this and throwing an error where it was previously constructed.

Reply:
@niloc132 What is `WhereNoneFilter` for?

Reply:
In DHE, SelectNoneFilter is a singletone added so that the "search" filter works. Search allows the client to provide a value to search against all columns (that can be matched anyway) at once. 

in DHC, it is called WhereNoneFilter.

In both cases, `WhereFilterFactory.expandQuickFilter()` is given the table's definition and the search string - if no columns can be matched, then an empty array of filters will be returned, and rather than have everything match (as `Table.where()` will return everything...), the decision was made to create a singleton that just removed everything.

We probably could have instead created a new empty table from the same definition, but that wasnt what was implemented at the time. I can't speak further for the decisions made in that code, I wasn't directly involved, but there might have been a reason to do that.

Reply:
So, we "filter" the table to produce an empty result. Seems like a lot of work to get there. I get it, though.